src/axom/core/Array.hpp

```diff
@@ -728,48 +837,65 @@ class Array
   }
   /// @}
 
-  T* m_data;
-  IndexType m_num_elements;
-  IndexType m_capacity;
-  double m_resize_ratio;
-  bool m_is_external;
+  T* m_data = nullptr;
+  /// \brief The full number of elements in the array
+  ///  i.e., 3 for a 1D Array of size 3, 9 for a 3x3 2D array, etc
+  IndexType m_num_elements = 0;
+  IndexType m_capacity = 0;
+  double m_resize_ratio = DEFAULT_RESIZE_RATIO;
+  bool m_is_external = false;
   int m_allocator_id;
+  // FIXME: What's the best way to make these available to users
+  // Is it sufficient to just give them the raw arrays?
+  /// \brief The sizes (extents?) in each dimension
+  IndexType m_dims[dim];
+  /// \brief The strides in each dimension
+  IndexType m_strides[dim];
```

Comment:
Is there a way to only create these when `dim > 1` ?

Reply:
What's the overhead for an array (in bytes)? 
Looks like 40+ bytes for 1D arrays (depending on if `IndexType` is 32 or 64 bits).

Perhaps we should try to reduce this, if possible in future revisions?

Reply:
Using 8 bytes to store resize_ratio as a `double` does seem kind of wasteful, do we really need 16 significant digits of precision for that? I've never really wanted to change the resize ratio on `std::vector`. In the 1D case, we shouldn't need more than `T * ptr, size, capacity`.

What does `m_is_external` do? The strides array doesn't need to exist in the 1D case.